airbyte-integrations/connectors/source-chargebee/unit_tests/integration/test_addon.py

```diff
@@ -0,0 +1,198 @@
+    return create_record_builder(
+        find_template(_STREAM_NAME, __file__),
+        FieldPath("list"),
+        record_id_path=NestedPath([_STREAM_NAME, _PRIMARY_KEY]),
+        record_cursor_path=NestedPath([_STREAM_NAME, _CURSOR_FIELD])
+    )
+
+def _a_response() -> HttpResponseBuilder:
+    return create_response_builder(
+        find_template(_STREAM_NAME, __file__),
+        FieldPath("list"),
+        pagination_strategy=ChargebeePaginationStrategy()
+    )
+
+def _read(
+    config_builder: ConfigBuilder,
+    sync_mode: SyncMode,
+    state: Optional[Dict[str, Any]] = None,
+    expecting_exception: bool = False
+) -> EntrypointOutput:
+    catalog = _catalog(sync_mode)
+    config = config_builder.build()
+    return read(_source(), config, catalog, state, expecting_exception)
+
+@freezegun.freeze_time(_NOW.isoformat())
+class FullRefreshTest(TestCase):
+
+    def setUp(self) -> None:
+        self._now = _NOW
+        self._now_in_seconds = int(self._now.timestamp())
+        self._start_date = _NOW - timedelta(days=28)
+        self._start_date_in_seconds = int(self._start_date.timestamp())
+
+    @staticmethod
+    def _read(config: ConfigBuilder, expecting_exception: bool = False) -> EntrypointOutput:
+        return _read(config, SyncMode.full_refresh, expecting_exception=expecting_exception)
+
+    @HttpMocker()
+    def test_given_one_page_of_records_read_and_returned(self, http_mocker: HttpMocker) -> None:
+        # Tests simple read
+        http_mocker.get(
+            _a_request().with_any_query_params().build(),
+            _a_response().with_record(_a_record()).with_record(_a_record()).build()
+        )
+        output = self._read(_config().with_start_date(self._start_date))
+        assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_multiple_pages_of_records_read_and_returned(self, http_mocker: HttpMocker) -> None:
+        # Tests pagination
+        request = _a_request().with_any_query_params().build()
+        http_mocker.get(
+            request,
+            [_a_response().with_record(_a_record()).with_pagination().build(), _a_response().with_record(_a_record()).build()]
+        )
+        output = self._read(_config().with_start_date(self._start_date - timedelta(hours=8)))
+        http_mocker.assert_number_of_calls(request, 2)
+        assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_http_status_400_when_read_then_stream_is_ignored(self, http_mocker: HttpMocker) -> None:
+        # Tests 400 status error handling
+        http_mocker.get(
+            _a_request().with_any_query_params().build(),
+            a_response_with_status(400)
+        )
+        output = self._read(_config().with_start_date(self._start_date), expecting_exception=True)
+        assert len(output.get_stream_statuses(f"{_STREAM_NAME}s")) == 0
+
+
+    @HttpMocker()
+    def test_given_http_status_401_when_the_stream_is_incomplete(self, http_mocker: HttpMocker) -> None:
```

Comment:
Can we be more aligned with the domain here? I feel it would be very helpful to understand in which case the API returns a 401. It usually relates to authentication issues but this test mentions it is a system_error and not a config_error

Reply:
Yep. Having a test is a step in the right direction, but I'd love to enforce that permission-related errors should in general be config errors.

Reply:
@maxi297 I confirmed that what is expected is a config_error as it is a non-actionable error.(https://github.com/airbytehq/airbyte/pull/35129)